tests/unit/authorizers/test_basic_authorizer.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+import base64
+import pytest
+import six
+
+from globus_sdk.authorizers import BasicAuthorizer
+
+USERNAME = "testUser"
+PASSWORD = "PASSWORD"
+
+
+@pytest.fixture
+def authorizer():
+    return BasicAuthorizer(USERNAME, PASSWORD)
+
+
+def test_set_authorization_header(authorizer):
+    """
+    Sets authorization header in a test dictionary, confirms expected value
+    """
+    header_dict = {}
+    authorizer.set_authorization_header(header_dict)
+    assert header_dict["Authorization"][:6] == "Basic "
+    decoded = base64.b64decode(
+        six.b(header_dict["Authorization"][6:])).decode('utf-8')
+    assert decoded == "{}:{}".format(USERNAME, PASSWORD)
+
+
+def test_set_authorization_header_existing(authorizer):
+    """
+    Confirms that an existing Authorization field is overwritten
+    """
+    header_dict = {"Header": "value",
+                   "Authorization": "previous_value"}
+    authorizer.set_authorization_header(header_dict)
+    assert header_dict["Authorization"][:6] == "Basic "
+    decoded = base64.b64decode(
+        six.b(header_dict["Authorization"][6:])).decode('utf-8')
+    assert decoded == "{}:{}".format(USERNAME, PASSWORD)
+    assert header_dict["Header"] == "value"
+
+
+def test_handle_missing_authorization(authorizer):
+    """
+    Confirms that BasicAuthorizer doesn't handle missing authorization
+    """
+    assert not authorizer.handle_missing_authorization()
+
+
+@pytest.mark.parametrize(
+    'username, password',
+    [("user", u"テスト"), (u"дум", 'pass'), (u"テスト", u"дум")])
+def test_unicode_handling(username, password):
```

Comment:
I like testing non ascii characters here, is there a good way to make sure we're testing it across the whole SDK?

Reply:
I think sprinkling some non ascii unicode values into the fixtures is probably sufficient, I mostly want to know that in our handling of both python 2 and 3 we aren't choking on utf8 conversions somewhere in the SDK itself.